### PR TITLE
Add fiber section to configuration reference

### LINKED ISSUE #4017: [Config] Document 'fiber' configuration settings
**Product:** Tarantool
**Since:** 3.0
**Root document:** 
- A new `fiber` section in the [Configuration reference](https://www.tarantool.io/en/doc/latest/reference/configuration/configuration_reference/).

**SME:** @ Totktonada

# Details

Options:
- https://github.com/tarantool/tarantool/blob/f58bfc97c627f1838696697339fb3e443c05767a/src/box/lua/config/instance_config.lua#L702

Related topics:
- https://www.tarantool.io/en/doc/latest/concepts/coop_multitasking/
- https://www.tarantool.io/en/doc/latest/reference/reference_lua/fiber/

### PR DESCRIPTION
**Pull request by lenkis**

### CODE REVIEW

File: doc/reference/configuration/configuration_reference.rst [hunk 1]
```diff
@@ -1249,6 +1217,148 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+The ``fiber`` section describes options related to configuring :ref:`fibers, yields, and cooperative multitasking <concepts-coop_multitasking>`.
+
+..  NOTE::
+
+    ``fiber`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+* :ref:`fiber.io_collect_interval <configuration_reference_fiber_io_collect_interval>`
+* :ref:`fiber.too_long_threshold <configuration_reference_fiber_too_long_threshold>`
+* :ref:`fiber.worker_pool_threads <configuration_reference_fiber_worker_pool_threads>`
+* :ref:`fiber.slice.* <configuration_reference_fiber_slice_options>`
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst:1234]: Rendered as a plain text, not a hyperlink.

File: doc/reference/configuration/configuration_reference.rst [hunk 2]
```diff
@@ -1249,6 +1217,148 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+fiber.slice.*
+~~~~~~~~~~~~~
+
+This section describes options related to configuring
+:ref:`fiber slices <fibers_limit_execution_time>`.
+
+*   :ref:`fiber.slice <configuration_reference_fiber_slice>`
+
+    * :ref:`fiber.slice.warn> <configuration_reference_fiber_slice_warn>`
+    * :ref:`fiber.slice.warn> <configuration_reference_fiber_slice_err>`
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst]: 1. Both bullets say 'warn' -- one should be about 'err'.
2. Extra `>` symbols.

File: doc/reference/configuration/configuration_reference.rst [hunk 3]
```diff
@@ -1249,6 +1217,148 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+
+fiber.top.*
+~~~~~~~~~~~
+
+This section describes options related to configuring the
+:ref:`fiber.top() <fiber-top>` function, normally used for debug purposes.
+
+*   :ref:`fiber.top <configuration_reference_fiber_top>`
+
+    * :ref:`fiber.top.enabled> <configuration_reference_fiber_top_enabled>`
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst]: Extra `>` symbol.

image

File: doc/reference/configuration/configuration_reference.rst [hunk 4]
```diff
@@ -1249,6 +1217,148 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+*   :ref:`fiber.slice <configuration_reference_fiber_slice>`
+
+    * :ref:`fiber.slice.warn> <configuration_reference_fiber_slice_warn>`
+    * :ref:`fiber.slice.warn> <configuration_reference_fiber_slice_err>`
+
+..  _configuration_reference_fiber_slice:
+
+..  confval:: fiber.slice
+
+    Set timings for all types of fiber slices.
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst]: We can link a function that is called under the hood. For example,

> See [fiber.set_max_slice](https://www.tarantool.io/en/doc/latest/reference/reference_lua/fiber/#fiber-set-max-slice) for details and examples.

**Review by Totktonada — APPROVED** [review]

File: doc/reference/configuration/configuration_reference.rst (hunk 2 above)
Comment (lenkis) [reply doc/reference/configuration/configuration_reference.rst]: Thanks!

File: doc/reference/configuration/configuration_reference.rst (hunk 3 above)
Comment (lenkis) [reply doc/reference/configuration/configuration_reference.rst]: Thanks!

File: doc/reference/configuration/configuration_reference.rst (hunk 4 above)
Comment (lenkis) [reply doc/reference/configuration/configuration_reference.rst]: Done.

File: doc/reference/configuration/configuration_reference.rst [hunk 5]
```diff
@@ -1249,6 +1217,149 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+* :ref:`fiber.too_long_threshold <configuration_reference_fiber_too_long_threshold>`
+* :ref:`fiber.worker_pool_threads <configuration_reference_fiber_worker_pool_threads>`
+* :ref:`fiber.slice.* <configuration_reference_fiber_slice_options>`
+* :ref:`fiber.top.* <configuration_reference_fiber_top_options>`
+
+..  _configuration_reference_fiber_io_collect_interval:
+
+..  confval:: fiber.io_collect_interval
+
+    The :ref:`fiber <app-fibers>` will sleep for the defined number of seconds
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]: If possible, I'd use the same pattern for all config option descriptions. Smth like:

_The time period (in seconds) a fiber sleeps between iterations of the event loop._

File: doc/reference/configuration/configuration_reference.rst [hunk 6]
```diff
@@ -1249,6 +1217,149 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+    between iterations of the event loop. Can be used to reduce CPU load in deployments
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]: The sentence looks incomplete if starts with `Can`:

```suggestion
    between iterations of the event loop. ``io_collect_interval`` can be used to reduce CPU load in deployments
```

File: doc/reference/configuration/configuration_reference.rst [hunk 7]
```diff
@@ -1249,6 +1217,149 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+    | Default: box.NULL
+    | Environment variable: TT_FIBER_IO_COLLECT_INTERVAL
+
+..  _configuration_reference_fiber_too_long_threshold:
+
+..  confval:: fiber.too_long_threshold
+
+    If processing a request takes longer than the given value (in seconds),
+    the fiber warns about it in the log.
+    Has effect only if :ref:`log.level <configuration_reference_log_level>`
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]:
```suggestion
    ``too_long_threshold`` has effect only if :ref:`log.level <configuration_reference_log_level>`
```

File: doc/reference/configuration/configuration_reference.rst [hunk 8]
```diff
@@ -1249,6 +1217,149 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+    | Type: number
+    | Default: 0.5
+    | Environment variable: TT_FIBER_TOO_LONG_THRESHOLD
+
+..  _configuration_reference_fiber_worker_pool_threads:
+
+..  confval:: fiber.worker_pool_threads
+
+    The maximum number of threads to use during execution
+    of certain internal processes (currently
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]: I'd use `for example` to protect us from possible changes here. Developers can apply this limit to other processes and say nothing to us.

```suggestion
    of certain internal processes (for example,
```

File: doc/reference/configuration/configuration_reference.rst [hunk 9]
```diff
@@ -1249,6 +1217,149 @@ The ``database`` section defines database-specific configuration parameters, suc
...
+:ref:`fiber slices <fibers_limit_execution_time>`.
+
+*   :ref:`fiber.slice <configuration_reference_fiber_slice>`
+
+    * :ref:`fiber.slice.warn <configuration_reference_fiber_slice_warn>`
+    * :ref:`fiber.slice.err <configuration_reference_fiber_slice_err>`
+
+..  _configuration_reference_fiber_slice:
+
+..  confval:: fiber.slice
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]: Let's drop this option completely. Having the `fiber.slice.*` section is enough to group slice options. Also, there is no `TT_FIBER_SLICE` env variable:

image

The same for `fiber.top`:

image

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

**Review by andreyaksenov — APPROVED** [review]

**Pull request merged**